### PR TITLE
DT-2229 latest queue lib

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/config/HmppsSnsProperties.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.ConstructorBinding
+
+const val LOCALSTACK_ARN_PREFIX = "arn:aws:sns:eu-west-2:000000000000:"
+
+@ConstructorBinding
+@ConfigurationProperties(prefix = "hmpps.sns")
+@ConditionalOnProperty(name = ["hmpps.sqs.provider"], havingValue = "localstack")
+data class HmppsSnsProperties(val topics: Map<String, TopicConfig>) {
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/config/HmppsSnsProperties.kt:12]: I'm moving these to the queue library soon.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/config/LocalstackSqsConfig.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+@Configuration
+@ConditionalOnProperty(name = ["hmpps.sqs.provider"], havingValue = "localstack")
+class LocalstackSqsConfig {
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @Bean
+  @DependsOn("hmppsQueueService")
+  fun awsSnsClient(
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/config/LocalstackSqsConfig.kt:26]: Moving to queue library soon.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/IndexQueueService.kt [hunk 3]
```diff
@@ -1,50 +1,48 @@
...
+  private val indexQueue = hmppsQueueService.findByQueueId("indexqueue") ?: throw MissingQueueException("HmppsQueue indexqueue not found")
+
+  private val indexQueueSqsClient = indexQueue.sqsClient as AmazonSQSAsync
+  private val indexQueueSqsDlqClient = indexQueue.sqsDlqClient as AmazonSQSAsync
+  private val indexQueueUrl: String = indexQueue.queueUrl
+  private val indexDlqUrl: String = indexQueue.dlqUrl
 
   fun sendIndexRequestMessage(payload: PrisonerIndexRequest) {
-    awsSqsIndexASyncClient.sendMessageAsync(SendMessageRequest(indexQueueUrl, gson.toJson(payload)))
+    indexQueueSqsClient.sendMessageAsync(SendMessageRequest(indexQueueUrl, gson.toJson(payload)))
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/IndexQueueService.kt:26]: Much of this PR involves standardizing the SQS client bean names throughout the project.

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**